packages/react/src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -58,18 +58,16 @@ export class FocusTrapZone extends React.Component<FocusTrapZoneProps, {}> {
 
   componentDidMount(): void {
     FocusTrapZone._focusStack.push(this)
-    const { disableFirstFocus = false } = this.props
-
-    this._previouslyFocusedElementOutsideTrapZone = this._getPreviouslyFocusedElementOutsideTrapZone()
+    this._bringFocusIntoZone()
+    this._hideContentFromAccessibilityTree()
+  }
 
-    if (
-      !this._root.current.contains(this._previouslyFocusedElementOutsideTrapZone) &&
-      !disableFirstFocus
-    ) {
-      this._findElementAndFocusAsync()
+  componentDidUpdate(): void {
+    const activeElement = document.activeElement as HTMLElement
+    // if after componentDidUpdate focus is not inside the focus trap, bring it back
+    if (!this._root.current.contains(activeElement)) {
+      this._bringFocusIntoZone()
```

Comment:
I'm not sure this is a good idea. A component can update for many reasons and I am pretty sure it's not always expected behavior to bring focus back to the zone after every update.

Is see this behavior as risky because it can cause unexpected focus jumps so I'd recommend allowing the users to configure the focus zone to add / disable this behavior. Let's wait for other opinions.

Reply:
@Bugaa92 this will only work if the active element (currently focused element) is not inside the focus trap container (for any reason) - focus should also be there once it is rendered.

Reply:
as discusse doffline - one scenario where this might not work is if the popup opens another popup with focus trap. Then if the first popup updates, it might steal focus from the second popup. Can you please test this scenario?

Reply:
@jurokapsiar I tested this scenario. Since inner popup is rendered on button click in the first popup without relation to the outside state, it doesn't trigger componentDidUpdate

Reply:
@sophieH29 can we make this behavior configurable via prop?

Reply:
I've tried this case and it works
```jsx
<Popup> -- with lazy load state change    
      <InnerPopup> -- with lazy load state change
              <InnerInnerPopup> -- autocontrolled Popup
                      <Input>
```
fyi @Bugaa92 @jurokapsiar

Reply:
@layershifter agree, updated